bob-backend/src/pearl/holder.rs

```diff
@@ -256,20 +259,24 @@ impl Holder {
             let res = storage
                 .read(Key::from(key))
                 .await
-                .map(|r| {
-                    counter!(PEARL_GET_BYTES_COUNTER, r.len() as u64);
-                    BobData::from_serialized_bytes(r)
-                })
                 .map_err(|e| {
                     counter!(PEARL_GET_ERROR_COUNTER, 1);
                     trace!("error on read: {:?}", e);
-                    match e.downcast_ref::<PearlError>().unwrap().kind() {
-                        PearlErrorKind::RecordNotFound => Error::key_not_found(key),
-                        _ => Error::storage(e.to_string()),
+                    Error::storage(e.to_string())
+                })
+                .and_then(|r| match r {
+                    ReadResult::Found(v) => {
+                        counter!(PEARL_GET_BYTES_COUNTER, v.len() as u64);
+                        BobData::from_serialized_bytes(v.to_vec()).map(|d| ReadResult::Found(d))
```

Comment:
Don't forget to remove `v.to_vec()` because after request #637 the `Bytes` type will be used